src/mb_enhanced_cover_art_uploads/maximise.ts

```diff
@@ -293,60 +294,30 @@ IMU_EXCEPTIONS.set('img.discogs.com', async (smallurl) => {
 
 // Apple Music
 IMU_EXCEPTIONS.set('*.mzstatic.com', async (smallurl) => {
-    // For Apple Music, IMU always returns a PNG, regardless of whether the
-    // original source image was PNG or JPEG. We can grab the source image
-    // instead. See #80.
+    // For Apple Music, IMU always returns a `/source` URL first, but it may
+    // not always exist. Mark such a URL as likely broken so that we don't warn
+    // for those. We don't reorder the candidates to ensure we always get the
+    // largest possible image.
 
     const results: MaximisedImage[] = [];
+    let lastSource: MaximisedImage | undefined;
     for await (const imgGeneric of maximiseGeneric(smallurl)) {
-        // Assume original file name is penultimate component of pathname, e.g.
-        // https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/58/34/98/58349857-55bb-62ae-81d4-4a2726e33528/5060786561909.png/999999999x0w-999.png
-        // Transform it to the following:
-        // https://a1.mzstatic.com/us/r1000/063/Music124/v4/58/34/98/58349857-55bb-62ae-81d4-4a2726e33528/5060786561909.png
-        // i.e. change domain to a1, path prefix to us/r1000/063, and drop trailing conversion name.
-        // This should also work for images returned by the old iTunes API.
-        const sourceUrl = new URL(imgGeneric.url);
-        sourceUrl.hostname = 'a1.mzstatic.com';
-
-        // Be a bit defensive and don't try to transform URLs which we don't recognize
-        if (sourceUrl.pathname.startsWith('/image/thumb')) {
-            sourceUrl.pathname = sourceUrl.pathname.replace(/^\/image\/thumb/, '/us/r1000/063');
+        if (urlBasename(imgGeneric.url) === 'source') {
+            lastSource = imgGeneric;
+        } else if (
+            typeof lastSource !== 'undefined'
+                && urlDirname(imgGeneric.url) === urlDirname(lastSource.url)
+                && imgGeneric.url.hostname === lastSource.url.hostname) {
+            // Mark the `/source` image as likely broken if there's another URL
+            // with the same directory but a different image name.
+            lastSource.likely_broken = true;
```

Comment:
FWIW, https://a1.mzstatic.com/us/r1000/063/Music111/v4/e1/dc/68/e1dc6808-6d55-1e38-a34d-a3807d488859/191061355977.jpg still shows the "Skipping maximised candidate source: HTTP error 404: Not Found" warning as `source` is a 404 (example taken from https://github.com/qsniyg/maxurl/commit/56e0fce886bd328e8a155fc69dd62ff954fa0053). So this is not always working as expected?

Reply:
Hmm, for that URL IMU only returns one candidate, so it's never executing the iteration that would mark it as `likely_broken`. Will rewrite the logic to not rely on multiple candidates, it should instead compare the maximised URL to the one that was given as input.

Reply:
Should be fixed now.